### PR TITLE
feat: new http agent for communication with dojot

### PR DESCRIPTION
**Pull request by gilmaicor**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)
* #2192 

* **Other information**:

### CODE REVIEW

**Review by luissajunior — CHANGES_REQUESTED** [review]

File: connector/http/app/express/interceptors/deviceIdentification.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        const { fingerprint256 } = clientCert;
+        let messageKey = cache.get(fingerprint256);
+        if (messageKey) {
+          [req.tenant, req.deviceId] = messageKey;
+          return next();
+        }
+        try {
+          messageKey = await axios
+            .get(
+              `http://certificate-acl:3000/internal/api/v1/acl-entries/${fingerprint256}`,
```
[the remark this answers is not in the document]
Comment (gilmaicor) [reply connector/http/app/express/interceptors/deviceIdentification.js:51]: It's an internal dojot url, there should be no change.

File: connector/http/app/express/interceptors/deviceIdentification.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            )
+            .then((resp) => resp.data.split(':'));
+          [req.tenant, req.deviceId] = messageKey;
+          cache.set(fingerprint256, messageKey, 30000);
```
[the remark this answers is not in the document]
Comment (gilmaicor) [reply connector/http/app/express/interceptors/deviceIdentification.js:55]: Done.

File: connector/http/app/express/interceptors/deviceIdentification.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        }
+      }
+
+      err.message = 'Client certificate is invalid';
+      return next(err);
+    }
+
+    const reqType = req.path.split('/');
+
+    if (config['unsecure.mode'] && reqType[3] === 'unsecure') {
```
[the remark this answers is not in the document]
Comment (gilmaicor) [reply connector/http/app/express/interceptors/deviceIdentification.js:70]: This is part of the url for insecure mode, there should be no change.

File: connector/http/app/express/routes/v1/IncomingMessages.js [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+                res.status(HttpStatus.BAD_REQUEST).json({
+                  success: false,
+                  message: JSON.parse(e.message),
+                });
+              }
+            },
+          ],
+        },
+      ],
+    },
```
[the remark this answers is not in the document]
Comment (gilmaicor) [reply connector/http/app/express/routes/v1/IncomingMessages.js:130]: Done.

**Pull request closed without merging**